### PR TITLE
Neaten error-checking

### PR DESCRIPTION
**Pull request by twaugh**
Signed-off-by: Tim Waugh <twaugh@redhat.com>

### CODE REVIEW

File: backvendor/vendored.go [hunk 1]
```diff
@@ -131,20 +131,13 @@ func updateHashesAfterStrip(hashes *FileHashes, wt *WorkingTree, ref string, pat
...
-			}()
-
-			_, err = w.WriteTo(f)
-			return err
-		}()
-
+		_, err = w.WriteTo(f)
 		if err != nil {
+			f.Close() // ignore any secondary error
+			return true, errors.Wrap(err, "updating hash")
```
Comment (mprahl) [inline backvendor/vendored.go]: Why are you returning `true` directly? Why is `anyChanged` not used here?

File: backvendor/vendored.go [hunk 2]
```diff
@@ -131,20 +131,13 @@ func updateHashesAfterStrip(hashes *FileHashes, wt *WorkingTree, ref string, pat
...
+		}
+
+		if err = f.Close(); err != nil {
 			return true, errors.Wrap(err, "updating hash")
```
Comment (mprahl) [inline backvendor/vendored.go]: Same as above

**Review by mprahl — APPROVED** [review]
:+1: since this doesn't actually change the logic, but I am confused as to what the boolean being returned signifies when there is an error.

**Comment by twaugh** [comment]
Yes, this was a bit fuzzy. I've tightened it up to indicate whether any of the supplied hashes were modified, and added documentation for the function.

**Review by mprahl — APPROVED** [review]
:+1:

**Pull request merged**